Mathlib/Tactic/Ring/RingNF.lean

```diff
@@ -249,6 +249,11 @@ example (n : ℕ) (m : ℤ) : 2^(n+1) * m = 2 * 2^n * m := by ring
 example (a b : ℤ) (n : ℕ) : (a + b)^(n + 2) = (a^2 + b^2 + a * b + b * a) * (a + b)^n := by ring
 example (x y : ℕ) : x + id y = y + id x := by ring!
 ```
+
+This can be used non-terminally to normalize ring expressions in the goal such as
```

Comment:
What's the referent of "This" here? It doesn't seem like it is the `ring` tactic, since the following sentence says it can be used to prove things that `ring` cannot.

Since this text was copied from the module docstring, maybe it applies to `ring_nf` instead? But the module docstring is also confusing to me since it states that `ring_nf` uses `ring`, but as far as I can tell, it's actually the other way around?

cc: @digama0

Reply:
I agree. There should be some fixes. Let's use this PR for it.

Reply:
Okay, moved the passage to the `ring_nf` docstring. Thanks.

Reply:
@bryangingechen `ring_nf` uses `ring` infrastructurally (`ring_nf` is the more complicated tactic, built on `ring`), but it also installs itself as a hook to `ring` so that `ring` will try `ring_nf` on failure and suggest that.